app/src/hooks/useProjectRootPath.ts

```diff
@@ -0,0 +1,21 @@
+import { useLocation, useParams } from "react-router";
+
+/**
+ * Returns the root path for a project url
+ * @example
+ * "/projects/123/spans/456" -> "/projects/123"
+ * @returns the root path for a project
+ */
+export const useProjectRootPath = () => {
+  const { projectId } = useParams();
+  if (!projectId) {
+    throw new Error("projectId is required");
+  }
+  const location = useLocation();
+  const rootPath = location.pathname
+    .split("/")
+    .slice(0, location.pathname.split("/").indexOf(projectId) + 1)
+    .join("/");
+  const tab = location.pathname.split(rootPath)[1].split("/")[1];
```

Comment:
blindly indexing into an array seems not quite ideal. Also the tab names could use a type guard to make things more error proof